### PR TITLE
[#12238] Instructors editing session dates: disable calendar popup if the form is not in edit mode

### LINKED ISSUE #12238: Instructors editing session dates: disable calendar popup if the form is not in edit mode
v8.25.0, production

Context: An instructor is in the Session edit page, and without clicking the `edit`  button on top-right, clicks on the calendar icon.
image

Problem: The calendar pops up with all dates disabled. This can make the user thinks the form is broken.

Suggested: Disable the button itself until the `Edit` button is clicked, similar to how the time selection dropdown works.

### PR DESCRIPTION
**Pull request by pgfrank1**
Calendar button is now disabled when the Instructor is out of the editing view for Sessions

Author: pgfrank1

Fixes #12238 

**Outline of Solution**
I updated the datepicker component to use isDisabled instead of disabled to keep consistency with other components. I then applied that change to the button component so the calendar cannot be opened.

image

### CODE REVIEW

File: src/web/app/components/datepicker/datepicker.component.ts [hunk 1]
```diff
@@ -16,7 +16,7 @@ export class DatepickerComponent {
   date: DateFormat | undefined;
 
   @Input()
-  disabled: boolean = false;
+  isDisabled: boolean = false;
```
Comment (samuelfangjw) [inline src/web/app/components/datepicker/datepicker.component.ts:19]: Thanks for cleaning this up!

**Review by samuelfangjw — APPROVED** [review]
LGTM, thanks for your first contribution to TEAMMATES!

**Comment by pgfrank1** [comment]
For one of the failing tests, do I also need to add the changes to the snapshot of the instructor session edit page? Such as:

`<button`
    `aria-label="Change date"`
    `class="btn btn-light input-group-text"`
    `disabled=""`
    `type="button"`
  `>`

 Also when I run these test locally I noticed that I get:

`__ngContext__={[Function Number]}`
`__ngContext__="0"`

Instead of:
`__ngContext__={[Function Number]}`

What is the ngContext referring to?

**Comment by samuelfangjw** [comment]
@pgfrank1, yes please update the jest snapshots. 

> Also when I run these test locally I noticed that I get:
> 
> `__ngContext__={[Function Number]}` `__ngContext__="0"`
> 
> Instead of: `__ngContext__={[Function Number]}`

Try running all the tests at once instead of filtering/running only failed tests. First test in the test suite is assigned `__ngContext__="0"`, other tests are assigned higher numbers that are casted into their primitive wrapper types for some odd reason.

> What is the ngContext referring to?

Some internal state used by angular, I'm not familiar with the specifics unfortunately.

**Review by zhaojj2209 — APPROVED** [review]
LGTM

**Pull request merged**